services/QuillLMS/engines/evidence/app/controllers/evidence/research/gen_ai/datasets_controller.rb

```diff
@@ -36,9 +45,20 @@ def show
           end
         end
 
+        private def create_dataset_from_file_for_json_request
+          @dataset = stem_vault.datasets.new(dataset_params)
```

Comment:
You'll need to add task_type into this initializer.  

There's no way you would have known this as the change was just merged in from my RAG stuff.  (see line 38 above)

Something like:
`@dataset = stem_vault.datasets.new(dataset_params.merge(task_type: Dataset::GENERATIVE))`